### PR TITLE
google-cloud-securitycenter python lib

### PR DESCRIPTION
**Pull request by rsojitra-crest**
## Status
In Progress

## Description
google-cloud-securitycenter lib for Demisto/Cortex XSOAR integration

### CODE REVIEW

**Comment by crestdatasystems** [comment]
googleapis/python-securitycenter is licensed under the [ Apache License 2.0](https://github.com/googleapis/python-securitycenter/blob/master/LICENSE)

File: docker/google-cloud-securitycenter/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+FROM demisto/python3:3.8.6.14516
+
+COPY requirements.txt .
+
+RUN apk --update add --no-cache --virtual .build-dependencies python3-dev build-base wget git \
+  && pip install --no-cache-dir -r requirements.txt \
+  && apk del .build-dependencies
```
Comment (bakatzir) [inline docker/google-cloud-securitycenter/Dockerfile:8]:
```suggestion
RUN apk --update add --no-cache --virtual .build-dependencies python3-dev build-base wget git \ gcc
  && pip install --no-cache-dir -r requirements.txt \
  && apk del .build-dependencies
```

Comment (glicht) [reply docker/google-cloud-securitycenter/Dockerfile:8]: gcc is installed as part of python3-dev. The problem the package is failing to compile not a missing compiler.

File: docker/google-cloud-securitycenter/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+FROM demisto/python3:3.8.6.14516
+
+COPY requirements.txt .
+
+RUN apk --update add --no-cache --virtual .build-dependencies python3-dev build-base wget git \
```
Comment (glicht) [inline docker/google-cloud-securitycenter/Dockerfile:6]: Need to add `linux-headers` so the install compiles.
```suggestion
RUN apk --update add --no-cache --virtual .build-dependencies python3-dev build-base wget git linux-headers \
```

**Review by glicht — CHANGES_REQUESTED** [review]

File: docker/google-cloud-securitycenter/Dockerfile (hunk 2 above)
Comment (glicht) [reply docker/google-cloud-securitycenter/Dockerfile:6]: Seems to be same as reported here: https://github.com/grpc/grpc/issues/21918

**Comment by crestdatasystems** [comment]
Hi @bakatzir / @glicht 
We have tested the above docker image file and getting the following error while testing

```
Script failed to run: 
Error: [Traceback (most recent call last):
   
   File "<string>", line 3, in <module>
   File "/usr/local/lib/python3.8/site-packages/google/cloud/securitycenter/__init__.py", line 18, in <module>
    from google.cloud.securitycenter_v1.services.security_center.async_client import (
   File "/usr/local/lib/python3.8/site-packages/google/cloud/securitycenter_v1/__init__.py", line 18, in <module>
    from .services.security_center import SecurityCenterClient
   File "/usr/local/lib/python3.8/site-packages/google/cloud/securitycenter_v1/services/security_center/__init__.py", line 18, in <module>
    from .client import SecurityCenterClient
   File "/usr/local/lib/python3.8/site-packages/google/cloud/securitycenter_v1/services/security_center/client.py", line 27, in <module>
    from google.api_core import gapic_v1  # type: ignore
   File "/usr/local/lib/python3.8/site-packages/google/api_core/gapic_v1/__init__.py", line 18, in <module>
    from google.api_core.gapic_v1 import config
   File "/usr/local/lib/python3.8/site-packages/google/api_core/gapic_v1/config.py", line 23, in <module>
    import grpc
   File "/usr/local/lib/python3.8/site-packages/grpc/__init__.py", line 23, in <module>
    from grpc._cython import cygrpc as _cygrpc
 ImportError: Error loading shared library libstdc++.so.6: No such file or directory (needed by /usr/local/lib/python3.8/site-packages/grpc/_cython/cygrpc.cpython-38-x86_64-linux-gnu.so)
] (2604) (2603)
```

We have resolved this issue by adding "RUN apk add --no-cache libstdc++" line in Dockerfile.
(Ref link: https://hub.docker.com/r/dahanna/python.3.7-pandas-alpine/dockerfile)

Please review the changes.

File: docker/google-cloud-securitycenter/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+FROM demisto/python3:3.8.6.14516
+
+COPY requirements.txt .
+
+RUN apk add --no-cache libstdc++
```
Comment (glicht) [inline docker/google-cloud-securitycenter/Dockerfile:6]: Please add --update. To make sure we install the  latest and greatest.

**Comment by glicht** [comment]
@crestdatasystems  please see my previous comment

File: docker/google-cloud-securitycenter/Dockerfile (hunk 3 above)
Comment (crestdatasystems) [reply docker/google-cloud-securitycenter/Dockerfile]: Thank you @glicht for the review comment. We have added --update for libstdc++ library.

**Review by glicht — APPROVED** [review]

**Pull request merged**